src/emulator/pubsubEmulator.ts

```diff
@@ -16,14 +19,19 @@ export interface PubsubEmulatorArgs {
   auto_download?: boolean;
 }
 
+interface Trigger {
+  triggerKey: string;
+  signatureType: SignatureType;
+}
+
 export class PubsubEmulator implements EmulatorInstance {
   pubsub: PubSub;
 
   // Map of topic name to a list of functions to trigger
-  triggers: Map<string, Set<string>>;
+  triggersByTopic: Map<string, { triggerKeys: Set<string>; triggers: Trigger[] }>;
```

Comment:
Please change to `triggersByTopicAndKey: Map<string, Map<string, Trigger>>`. Having to keep the key set in sync with array is annoying and may lead to edge cases.

In addition, consider dropping `Trigger#triggerKey` if not needed to reduce duplication. You can just loop around the inner map `entries`.

Reply:
Great suggestion. Simplified the data structure and renamed the variable.